### DISCUSSION TITLE
How to detach replication link after ETL transfer to other RavenDB DB

### DISCUSSION THREAD

**Discussion by chris-fellows** [ravendb/ravendb#15122] [Q&A]
RavenDB 5.4.2 (AWS cluster) - Docs are created via Dot NET Core
I've created an ETL script task to copy documents from the source DB to a RavenDB archive DB some hours before they're due to expire. At the same time then I increase the expiry time in the archive DB. However when the document is removed from the source DB (Expires/Manually deleted) then the document is deleted from the archive DB too.

How do I detach the link and stop the document in the archive DB from being deleted? I've tried to delete the @flags metadata and also set it to null or empty string but it does not work. The @flags value is always 'FromReplication'.

This is what I am specifying for deleting the metadata:
`delete this['@metadata']['@flags'];`

**Comment by ayende** [comment]
What is likely happening is that the delete is flowing to the other side as well.

(Assuming that you haven't enabled expiration on the destination db as well).

You can stop the ETL from sending deletes using `deleteDocumentsOf{CollectionName}Behavior` (so `deleteDocumentsOfUsersBehavior`, `deleteDocumentsOfOrdersBehavior`, etc.

See here: 

https://ravendb.net/docs/article-page/5.4/csharp/server/ongoing-tasks/etl/raven#deletions

In particular, this discuss how you can do that:
https://ravendb.net/docs/article-page/5.4/csharp/server/ongoing-tasks/etl/raven#deletions-generic-function

I assume that there are no updates, so you can add:

```
function deleteDocumentsBehavior(docId, collection, deleted) {
  return false; // don't send any deletes to the other side
}
```

**Reply by chris-fellows** [reply]
Thank you. If I add the deleteDocumentsBehavior function then the source doc is correctly copied to the archive DB after the refresh time is reached and I can now delete the source doc and the archived doc doesn't get deleted.

However the archive doc has the source doc expiry time but it needs to be increased. If I try and increase the expiry time before calling loadTo[Doc](this) then the document never appears in the archive DB. It's the same if I try to delete the @expiry metadata. If I remove deleteDocumentsBehavior then I can increase the expiry but obviously I will have the original issue when I delete the source doc.

How do I increase the archive doc expiry time? (The source doc has a 3 month expiry and the archive doc needs a 2 year expiry).

**Reply by ayende** [reply]
What does your *full* script looks like?
If you are writing to another collection, it may issue a delete & update, and you need to respond to that.

**Reply by chris-fellows** [reply]
For testing then I create a TestDoc doc with an expiry of 10 mins and a refresh 8 mins before expiry. The script will copy to archive DB when less than 9 mins to expiry. I need to increase the expiry of the archive doc to 5 yrs. The collection name in the archive DB doesn't have to be the same as the source DB.

```
function deleteDocumentsBehavior(docId, collection, deleted) {
    return false; // don't send any deletes to the other side
}

var expiry = new Date(this['@metadata']['@expires']);

// Calculate time prior to expiry when doc should be archived
var expirySecsBefore = (9 * 60);      // Time before expiry when doc should be archived
var keepTimeDays = 1825;    // 5 yrs
var now = new Date();
var expiryTimeFrom = new Date();
expiryTimeFrom.setTime(expiry.getTime() - (expirySecsBefore * 1000));

if (now >= expiryTimeFrom)   // Within period before expiry
{
    // This doesn't work. Setting expiry causes doc not to be copied
    // Removing this will work but expiry will be same as source DB
    expiry.setDate(expiry.getDate() + keepTimeDays);
    this['@metadata']['@expires'] =  expiry;

    // This doesn't work either. Causes doc not to be copied
    //delete this['@metadata']['@expires'];
    
    loadToTestDocs (this);
}

```

**Reply by garayx** [reply]
There is a bug when editing the metadata inside the ETL script: I have opened an issue to fix it: https://issues.hibernatingrhinos.com/issue/RavenDB-19466/Cannot-modify-metadata-in-ETL-script?s=Cannot-modify-metadata-in-ETL-script

Meanwhile you can use patch on the destination database to modify the `@expires` value
